src/wallet/rpcwallet.cpp

```diff
@@ -3300,10 +3300,29 @@ UniValue getstakereport(const UniValue& params, bool fHelp)
 
     vStakePeriodRange_T::iterator vIt;
 
+    // Span of days to compute average over
+    int nDays = 0;
+
     // report it
     for(vIt = aRange.begin(); vIt != aRange.end(); vIt++)
     {
+        // Add it to results
         result.push_back(Pair(vIt->Name, FormatMoney(vIt->Total).c_str()));
+
+        // Get the nDays value
+        nDays = 0;
+        if (vIt->Name == "Last 7 Days")
+            nDays = 7;
+        else if (vIt->Name == "Last 30 Days")
+            nDays = 30;
+        else if (vIt->Name == "Last 365 Days")
+            nDays = 365;
+
+        // Check if we need to add the average
+        if (nDays > 0) {
+            // Add the Average
+            result.push_back(Pair(vIt->Name + " Avg", FormatMoney(vIt->Total / nDays).c_str()));
```

Comment:
Shouldn't the average be calculated only over the amount for days where a stake happened? If my wallet is 60 days old, it'll need 305 days to show an accurate number in "365 Avg", for example.

Reply:
So you mean calculate the average stake for the last X days where X Days is the number of days past since your first stake that is no older than 365 days?

Well, that would not really be a 365 day average, that would be more an average since staking.

The current formula would be 100% accurate if your wallet is 1 year old, but as you say it would be completely off if your wallet has only been staking for like 45 days.

I'll see what I can come up with for making the average more accurate.

I think I can calculate the wallet age based on first stake date and use it to adjust nDays value.

Reply:
I would define “X Avg” as the daily average received as rewards over the last X days of active staking.

Reply:
I think this should be clearly documented somewhere so people don't accidentally assume the averages are simple averages over the entire X timeframe regardless of initial staking activity.